presto-main/src/main/java/com/facebook/presto/sql/rewrite/MaterializedViewOptimizationRewrite.java

```diff
@@ -87,4 +99,48 @@ protected Node visitQuery(Query query, Void context)
             return query;
         }
     }
+
+    private static Query optimizeQueryUsingMaterializedView(
+            Metadata metadata,
+            Session session,
+            SqlParser sqlParser,
+            AccessControl accessControl,
+            Query node)
+    {
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(session, metadata);
+        materializedViewCandidateExtractor.process(node);
+        Map<QualifiedObjectName, List<QualifiedObjectName>> baseTableToMaterializedViewNameMap = materializedViewCandidateExtractor.getMaterializedViewCandidatesForTable();
+        // TODO: Select the most compatible and efficient materialized view for query rewrite optimization https://github.com/prestodb/presto/issues/16431
+        // TODO: Refactor query optimization code https://github.com/prestodb/presto/issues/16759
+
+        Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViewDefinitionMap = new HashMap<>();
```

Comment:
Suggestion: `ImmutableMap`

Reply:
Using a `Builder` here is non-trivial because we need to use a `get` operation before the map is fully constructed. I re-assigned using `ImmutableMap.copyOf()` instead.